### PR TITLE
fix: update redirectURL to redirect at new UI

### PR DESCRIPTION
**Pull request by pratikjagrut**
## Issue: 

 https://github.com/rancher/rancher/issues/47801

## Problem

After logging in, hit a 500 error. Was taken to a screen that didn't match the newer look and feel.
https://jira.suse.com/browse/SURE-9109
 
## Solution

Update the URL in saml backend.
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: pkg/auth/providers/saml/saml_client.go [hunk 1]
```diff
@@ -363,6 +363,9 @@ func (s *Provider) HandleSamlAssertion(w http.ResponseWriter, r *http.Request, a
 	rancherAction := s.clientState.GetState(r, "Rancher_Action")
 
 	if rancherAction == loginAction {
+		if !strings.Contains(redirectURL, "/dashboard/auth") {
```
Comment (JonCrowther) [inline pkg/auth/providers/saml/saml_client.go:366]: Could you add `/dashboard/auth` and `/login?` as constants?

File: pkg/auth/providers/saml/saml_client.go [hunk 2]
```diff
@@ -394,7 +397,7 @@ func (s *Provider) HandleSamlAssertion(w http.ResponseWriter, r *http.Request, a
 	if err != nil {
 		log.Error(err)
 		// UI uses this translation key to get the error message
-		http.Redirect(w, r, redirectURL+"errorCode=422&errorMsg="+UITranslationKeyForErrorMessage, http.StatusFound)
+		http.Redirect(w, r, redirectURL+"errorCode=422&err="+UITranslationKeyForErrorMessage, http.StatusFound)
```
Comment (JonCrowther) [inline pkg/auth/providers/saml/saml_client.go:400]: What does this change affect? Is it causing the wrong re-direct too?

**Review by JonCrowther — COMMENTED** [review]
Could you add a unit test for `HandleSamlAssertion`?

File: pkg/auth/providers/saml/saml_client.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/auth/providers/saml/saml_client.go:366]: Okay

File: pkg/auth/providers/saml/saml_client.go (hunk 2 above)
Comment (pratikjagrut) [reply pkg/auth/providers/saml/saml_client.go:400]: It looks like the UI team changed the `errorMsg` key to `err`. So now, if we send the error message with `errorMsg`, it doesn’t show up on the UI, but if we send it as `err`, it works and displays fine.

Comment (JonCrowther) [reply pkg/auth/providers/saml/saml_client.go:402]: Awesome! Good catch!

**Comment by pratikjagrut** [comment]
> Could you add a unit test for `HandleSamlAssertion`?

@JonCrowther  The changes I made are very small for a unit test, but even if I write one, it seems complicated because I will have to mock the Saml client and ClientState for cookies, the Saml assertion, and a few other functions. Also, in the end, the function does `HTTP.Redirect,` which is supposed to be handled by mux. Mocking everything seems excessive for the changes. So is it possible to skip it? If not, then can you help me with the unit test?

**Review by JonCrowther — APPROVED** [review]
I do think we'll want unit tests eventually, but I understand this is a small change and probably not worth the effort.

LGTM

**Review by alegrey91 — APPROVED** [review]
LGTM

**Pull request merged**